File: Co-op/JavaScript/point_to_point.js
//- Function to calculate the Euclidean distance between two vectors p and q
function euclideanDistance(p, q) {
    return Math.sqrt(p.reduce((sum, pi, i) => sum + Math.pow(pi - q[i], 2), 0));
  }
  
//- Function to calculate pairwise distance
function pd(p, q, measure = null, func = null) {
    if (measure === 'euclidean' || measure === null) {
      return euclideanDistance(p, q);
    } else if (measure === 'customize' && func !== null) {
      return func(p, q);
    } else if (measure === 'customize' && func === null) {
      return euclideanDistance(p, q);
    } else {
      // Add your custom distance measures here
      // For example, for a "manhattan" measure:
      // return p.reduce((sum, pi, i) => sum + Math.abs(pi - q[i]), 0);
      return null;
    }
  }
  
//- Function to calculate weighted pairwise distance
function pd_w(p, q, weights, measure = null, func = null) {
    const weightedP = p.map((pi, i) => pi * weights[i]);
    const weightedQ = q.map((qi, i) => qi * weights[i]);
    return pd(weightedP, weightedQ, measure, func);
  }
  

File: Python/Co-op/measures/point_to_point.py
import numpy as np
from scipy.spatial.distance import pdist

#- Function to calculate pairwise distance
def pd(p, q, measure=None, function=None):
    if measure == 'euclidean' or measure == None:
        return pdist(np.vstack([p, q]))[0]
    elif measure == 'customize' and function != None:
        return function(p, q)
    elif measure == 'customize' and function == None:
        return pdist(np.vstack([p, q]))[0]
    else:
        return pdist(np.vstack([p, q]), measure)[0]

#- Function to calculate weighted pairwise distance
def pd_w(p, q, weights, measure=None, function=None):
    return pd(np.multiply(p, weights), np.multiply(q, weights), measure, function)